src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -133,6 +133,30 @@ public static decimal Abs(decimal value)
             return decimal.Abs(value);
         }
 
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double Abs(double value)
```

Comment:
The changes overall look correct to me. However, I'd like to better understand the need for this change as I wouldn't expect this to provide any diffs for existing platforms or even make new platform bring up easier.

On RyuJIT `Abs` is `[Intrinsic]` today and so the actual `CRT` call will essentially never happen (modulo for indirect calls; such as function pointers, delegates, or reflection). We will always emit the correct assembly sequence instead; namely what is effectively `Sse.And(Vector128.CreateScalarUnsafe(value), Vector128.CreateScalarUnsafe(0x7FFFFFFF).AsSingle()).ToScalar()` on x86/x64 and what is effectively `AdvSimd.Abs(Vector64.CreateScalarUnsafe(value)).ToScalar()` on ARM64. This compiles down to a single instruction in both cases.

I would expect that Mono JIT, Mono LLVM, and NativeAOT are doing relatively the same, but if not I would expect a different PR rather than one updating the managed implementation here.

Reply:
I made this PR in the same spirit as ILogB change which went in earlier for .NET 7. My understanding was that we generally want to reduce number of FCalls, especially when it has zero cost.

> or even make new platform bring up easier.

Do you mean it will rather make new platform bring up harder? Hard for me to imagine how, since this managed implementation is using a few BitConverter methods, which are also used by existing methods in Math.cs.

Reply:
Mono JIT (mini), Interp and LLVM all handle it as intrinsic. I guess NativeAOT is also fine since it uses RyuJIT.

Reply:
> Do you mean it will rather make new platform bring up harder?

No, I was just indicating that, as it appears to me, this is effectively a zero-diff PR (at least for expected real world use-cases).

That is, this PR doesn't make the code easier to read or maintain, it doesn't make new platform bring up easier, etc. The one place it will impact is `indirect-calls`, that is calls to `Math.Abs` that happen via `delegates`, `function pointers`, or `reflection`; but the overhead of the indirection is going to far out weigh the FCALL to native and so it's not even clear that it's beneficial for that niche case. If we were to care about indirect calls, I'd actually expect a different implementation, likely just using recursion like we do in HWIntrinsics to ensure "optimal" codegen.